File: perf.js
"use strict";
var N=8000000;
var LOOPS=50;
var now = require("performance-now")
function sum(a,b,elements)
{
	var t0 = now();
	var sum=0;
	for(var index=0;index<N;++index)
	{
		sum+= Math.sin(a[index])*Math.cos(b[index]);
	}
	var t1 = now();
	console.log("N="+N + " time(ms) =" + (t1 - t0).toFixed(3) + " sum="+sum);
}
var a = new Array(N);
var b = new Array(N);
for(var index=0;index<N;++index)
{
	a[index]=400+index;
	b[index]=500+index;
}
for(var index=0;index<LOOPS;++index)
	sum(a,b,N);

File: perf.cpp
#include "stdafx.h"
#include <vector>
#include <algorithm>
#include <iostream>
#include <numeric>
#include <iterator>
#include <chrono>
#define LOOPS 50
#define N 8000000

template <typename T, typename E>
void add(const std::vector<T>& a, const std::vector<T>& b, size_t elements){
	auto start = std::chrono::high_resolution_clock::now();
	E sum = 0;
	for (int index = 0; index < elements; ++index)
		sum += (sin(a[index]) * cos(b[index]));

	auto end = std::chrono::high_resolution_clock::now();
	auto time = std::chrono::duration_cast<std::chrono::microseconds>(end - start).count();
	std::cout << "N="<<N <<" time(us) =" << time << " sum=" << sum << std::endl;

}

int main(){
	using namespace std;
	vector<int> a(N);
	vector<int> b(N);

	std::iota(a.begin(), a.end(), 400);
	std::iota(b.begin(), b.end(), 500);
	for (int index = 0; index < LOOPS; ++index)
		add<int, double>(a, b, N);
	return 0;
}
